publisher/start.sh

```diff
@@ -34,7 +34,27 @@ EORESTORE
       pg_restore /dumps/db.backup -U postgres -d ${DB_NAME}
 EORESTORE
 
+    # we need to sanitize sensitive data first
+    docker run --name sanitizing_db_container -d --rm -v ${DUMPS_DIR}:/files \
+    -e POSTGRES_DB=${DB_NAME} -e POSTGRES_PASSWORD= postgres:10.5
+    # wait for "database system is ready to accept connections", hoping 4 sec is enough
+    sleep 4;
 
+    docker exec sanitizing_db_container psql -f /files/${backup_roles_file} -U postgres
+    docker exec sanitizing_db_container pg_restore /files/${backup_file} -U postgres -d ${DB_NAME}
+
+    cat <<-EOSANITIZE | (docker exec -i sanitizing_db_container sh -c "cat > sanitize.sql")
+          UPDATE emails
+          SET
+            email = regexp_replace(email, '(^[^@]+)', md5(email)),
+            name  = md5(name);
+EOSANITIZE
+
+    docker exec sanitizing_db_container psql -f sanitize.sql -U postgres ${DB_NAME}
+    docker exec sanitizing_db_container pg_dump -Fc -U postgres -f /files/${backup_file}.sanitized ${DB_NAME} > /dev/null
+    docker stop sanitizing_db_container
+
+    docker exec publishing_db_container cp /files/${backup_file}.sanitized /dumps/db.backup
```

Comment:
need to not forget to remove 31 line related code

Reply:
Done